pkg/controllers/uiplugin/compatibility_matrix_test.go

```diff
@@ -44,34 +47,60 @@ func TestCompatibilityMatrixMaxVersions(t *testing.T) {
 }
 
 func TestLookupImageAndFeatures(t *testing.T) {
+	stubbedFunc := func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error { return nil }
+
+	// Curry the acm version into the list function
+	acmVersionFunc := func(version string) func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+		return func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+			multiClusterHubList, ok := list.(*mchv1.MultiClusterHubList)
+			if !ok {
+				return fmt.Errorf("invalid list object type")
+			}
+			multiClusterHubList.Items = []mchv1.MultiClusterHub{
+				{
+					Spec: mchv1.MultiClusterHubSpec{},
+					Status: mchv1.MultiClusterHubStatus{
+						CurrentVersion: version,
+					},
+				},
+			}
+			return nil
+		}
+	}
+
 	for _, tc := range []struct {
 		pluginType       uiv1alpha1.UIPluginType
 		clusterVersion   string
+		listFunction     ListFunction
 		expectedKey      string
 		expectedErr      error
 		expectedFeatures []string
 	}{
 		{
 			pluginType:     uiv1alpha1.TypeDashboards,
 			clusterVersion: "4.10",
+			listFunction:   stubbedFunc,
```

Comment:
could we pass the ACM version here instead of a function?

Reply:
I wasn't able to move the check out of the reconciliation loop as the check was missing the reconciliation context. However, I did refactor a bit to have the function call happen a few levels upward so that this does only pass in the version string rather than a function.